pandas/core/arrays/datetimelike.py

```diff
@@ -43,7 +48,7 @@ def cmp_method(self, other):
         if isinstance(other, ABCDataFrame):
             return NotImplemented
 
-        if isinstance(other, (np.ndarray, ABCIndexClass, ABCSeries)):
+        if isinstance(other, (np.ndarray, ABCIndexClass, ABCSeries, cls)):
```

Comment:
I think at the time of evaluation `cls` is going to be `DatetimeLikeArrayMixin`, so this should allow comparisons between e.g. DatetimeArray vs TimedeltaArray (though not sure if this is tested).  Should other EAs make it through this check?

Reply:
DatetimeLikeArrayMixin is the closes to what we have now, so I'd prefer just doing that for now.

Though yes, we should probably include any array-like here I would think.

Reply:
This turns out to be un-hittable because the calling methods all do a length check before dispatching here.  These 3 lines are deleted in #24573.